numpy/core/src/multiarray/scalarapi.c

```diff
@@ -168,7 +168,7 @@ scalar_value(PyObject *scalar, PyArray_Descr *descr)
      * Use the alignment flag to figure out where the data begins
      * after a PyObject_HEAD
      */
-    memloc = (npy_intp)scalar;
+    memloc = (uintptr_t)scalar;
```

Comment:
Seems like this casting to signed was the problem (and I first thought it can't be because the denominator is stored as a `denom - 1` here making the `2` a `1`...

If the value is too large, it would be a negative which breaks the rounding (and also means that a previously aligned value looks unaligned).